### ISSUE TITLE
meta element with delay=0 and no redirect causes infinite loop in browser

### ISSUE THREAD

**Issue by dd8** [act-rules/act-rules.github.io#1115]
https://act-rules.github.io/rules/bc659a

The example below passes because delay=0.  But it  doesn't redirect to a new page, so it causes the browser to refresh the page continuously (every zero seconds) and makes the page unusable. 

With Firefox on a MacBook with a touch bar it causes the touch bar to flash which fails either 2.3.1 [Three Flashes or Below Threshold] or 2.3.2 [Three Flashes] depending on what's displayed in the touch bar.

This also causes the entire page  to flash if it has a coloured background. If you add `<body style="background-color: red;">` to the example below it fails 2.3.1 and could trigger a seizure.

```
<!DOCTYPE html>
<html lang="en">
<meta http-equiv="refresh" content="0" />
<title>Infinite Loop</title>   
<body>
   <h1>Infinite Loop</h1>
</body></html>
```

**Comment by dd8** [comment]
The meta refresh triggers 2 distinct actions depending on whether URL= is present in content attribute.

1) redirect to a new page (when URL= is present)
2) refresh the current page

Refresh triggers additional failures like flashing and focus issues - maybe these should be 2 different rules?

**Comment by dd8** [comment]
For redirects to another page the rule doesn't match the corresponding WCAG failure, which is triggered by any non-zero delay, and has no 20 hour upper limit:
https://www.w3.org/TR/WCAG20-TECHS/F40.html

For refreshes of the same page it also doesn't match the corresponding WCAG failure, which is triggered by any delay of zero or greater, and has no 20 hour upper limit:
https://www.w3.org/TR/WCAG20-TECHS/F41.html

**Comment by Jym77** [comment]
Hum, from what I understand, this does not (so far) invalidate the rule…
For example, the fact that a >20h delay still fails 2.2.4 is not an issue. If the rule fails, then 2.2.4 fails for sure; if the rule passes, then 2.2.4 may still fail as indicated by the "further testing is needed" in the accessibility mapping.

However, we can improve things… But likely by making new rules to improve coverage.
* There could be a very similar rule without the 20 hours limit. That rule would fail 2.2.4 and 3.2.5 (but not 2.2.1). (we usually focus on A and AA criteria, mostly because that is what legislation usually require, rules for AAA criteria are of course welcome, and in cases like that fairly easy to write). 
=> Would you like to write such a rule, adapting from the current one? It should be fairly easy to write (essentially just removing the 20 hours limit and updating a couple of examples).
* Concerning the flashing issue (of your first message), there could also be a rule checking specifically this and failing 2.3.1 and 2.3.2. I don't think we have any rule for these criteria so far, so it'd be nice to start having one.
(I agree that this rule might need update saying that there must also be an URL in the content, this is not technically a problem with the rule because we don't claim that passing the rule passes 2.2.1 or any SC ("further testing is needed"), but it can still be a nice improvement)
=> Would you like to try and write such a rule? It's probably going to be a bit harder than the other because it's further away from existing rule…

**Comment by WilcoFiers** [comment]
I'm in full agreement with Jean-Yves on this one. Yes, an infinite loop redirect is bad and could cause accessibility problems, but that does not fit under the success criterion this rule tests, so adding that in wouldn't make sense here. A new rule could be created for this. I'm not opposed to that, although I don't think this is something that actually happens on live websites.

**Comment by WilcoFiers** [comment]
Closing, since nobody has picked this up. If someone's still interested, feel free to create the PR.
